pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManagerTest.java

```diff
@@ -0,0 +1,142 @@
+
+  private HelixManager createMockHelixManager() {
+    HelixManager helixManager = mock(HelixManager.class);
+    when(helixManager.getClusterName()).thenReturn(clusterName);
+    return  helixManager;
+  }
+
+  private HelixAdmin createMockHelixAdmin() {
+    HelixAdmin helixAdmin = mock(HelixAdmin.class);
+    return helixAdmin;
+  }
+
+  private ZkHelixPropertyStore createMockPropertyStore() {
+    ZkHelixPropertyStore propertyStore = mock(ZkHelixPropertyStore.class);
+    return propertyStore;
+  }
+
+  private PinotHelixResourceManager createMockHelixResourceManager() {
+    PinotHelixResourceManager helixResourceManager = mock(PinotHelixResourceManager.class);
+    return helixResourceManager;
+  }
+
+
+  @Test
+  public void testKafkaAssignment() throws Exception {
+    final HelixAdmin helixAdmin = createMockHelixAdmin();
+    final HelixManager helixManager = createMockHelixManager();
+    final PinotHelixResourceManager helixResourceManager = createMockHelixResourceManager();
+    final ZkHelixPropertyStore propertyStore = createMockPropertyStore();
+
+    PinotLLCRealtimeSegmentManager.create(helixAdmin, helixManager, propertyStore, helixResourceManager);
+    PinotLLCRealtimeSegmentManager segmentManager = PinotLLCRealtimeSegmentManager.getInstance();
+
+    final String topic = "someTopic";
+    final String rtTableName = "table_REALTIME";
+    final int nPartitions = 4;
+    final int nReplicas = 2;
+    String[] instances = {server1, server2, server3};
+
+    // Populate 'partitionSet' with all kafka partitions,
+    // As we find partitions in the assigment, we will remove the partition from this set.
+    Set<Integer> partitionSet = new HashSet<>(nPartitions);
+    for (int i = 0; i < nPartitions; i++) {
+      partitionSet.add(i);
+    }
+
+    ZNRecord znRecord = segmentManager.assignKafkaPartitions(topic, nPartitions, Arrays.asList(instances), nReplicas);
+
+    Map<String, List<String>> assignmentMap = znRecord.getListFields();
+    Assert.assertEquals(assignmentMap.size(), nPartitions);
+    // The map looks something like this:
+    // {
+    //  "0" : [S1, S2],
+    //  "1" : [S2, S3],
+    //  "2" : [S3, S4],
+    // }
+    // Walk through the map, making sure that every partition (and no more) appears in the key, and
+    // every one of them has exactly as many elements as the number of replicas.
+    for (Map.Entry<String, List<String>> entry : assignmentMap.entrySet()) {
+      int p = Integer.valueOf(entry.getKey());
+      Assert.assertTrue(partitionSet.contains(p));
+      partitionSet.remove(p);
+      Assert.assertEquals(entry.getValue().size(), nReplicas);
+      // Make sure that we have unique server entries in the list for that partition
+      Set allServers = allServers();
+      for (String server : entry.getValue()) {
+        Assert.assertTrue(allServers.contains(server));
+        allServers.remove(server);
+      }
+      // allServers may not be empty here.
+    }
```

Comment:
Do we want to check if every server has the same number of segments assigned to it? It seems that this test would pass if one server is assigned no segments.